src/probnum/random_variables/_normal.py

```diff
@@ -257,14 +258,30 @@ def __init__(
             entropy=entropy,
         )
 
-    @cached_property
+    @property
     def cov_cholesky(self) -> _ValueType:
         """Cholesky factor :math:`L` of the covariance
         :math:`\\operatorname{Cov}(X) =LL^\\top`."""
-        if self._compute_cov_cholesky is None:
-            raise NotImplementedError
 
-        return self._compute_cov_cholesky()
+        if not self.cov_cholesky_is_precomputed:
+            self.precompute_cov_cholesky()
+        return self._cov_cholesky_values
+
+    def precompute_cov_cholesky(self, damping_factor=COV_CHOLESKY_DAMPING):
+        """Overwrites values of cov_cholesky with new Cholesky factors (careful: in-
+        place operation!)."""
+        self._cov_cholesky_values = self._compute_cov_cholesky(
+            damping_factor=damping_factor
+        )
+
+    @property
+    def cov_cholesky_is_precomputed(self) -> bool:
```

Comment:
easy-access flag to see whether the Cholesky factor is already cached.

`if Normal.cov_cholesky_is_precomputed` reads a bit better than `if Normal._cov_cholesky_values is not None`.